From a5211ff01342bee732288f59258372340f3f80d9 Mon Sep 17 00:00:00 2001
From: Lionel Debieve <lionel.debieve@st.com>
Date: Tue, 15 Sep 2020 09:39:14 +0200
Subject: [PATCH 0182/1695] dt-bindings: add the STM32MP2 clock and reset
 bindings

Add the associated bindings for device tree and drivers.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git e19129b11a23b9dc9d5e72229f737dd7697b0ca8 dt-bindings: add the STM32MP2 clock and reset bindings]
Change-Id: I434fdecda8b31245fc8a9591042f037866c6bc2b
Signed-off-by: Lionel Debieve <lionel.debieve@st.com>
---
 core/include/drivers/stm32mp_dt_bindings.h    |   6 +
 .../dt-bindings/clock/stm32mp25-clks.h        | 492 ++++++++++++++++++
 .../dt-bindings/clock/stm32mp25-clksrc.h      | 226 ++++++++
 .../dt-bindings/reset/stm32mp25-resets.h      | 167 ++++++
 .../bindings/clock/st,stm32mp25-rcc.yaml      | 448 ++++++++++++++++
 5 files changed, 1339 insertions(+)
 create mode 100644 core/include/dt-bindings/clock/stm32mp25-clks.h
 create mode 100644 core/include/dt-bindings/clock/stm32mp25-clksrc.h
 create mode 100644 core/include/dt-bindings/reset/stm32mp25-resets.h
 create mode 100644 documentation/devicetree/bindings/clock/st,stm32mp25-rcc.yaml

diff --git a/core/include/drivers/stm32mp_dt_bindings.h b/core/include/drivers/stm32mp_dt_bindings.h
index 2ad61b69c..62d2a0112 100644
--- a/core/include/drivers/stm32mp_dt_bindings.h
+++ b/core/include/drivers/stm32mp_dt_bindings.h
@@ -19,4 +19,10 @@
 #include <dt-bindings/reset/stm32mp1-resets.h>
 #endif
 
+#ifdef CFG_STM32MP25
+#include <dt-bindings/clock/stm32mp25-clks.h>
+#include <dt-bindings/clock/stm32mp25-clksrc.h>
+#include <dt-bindings/reset/stm32mp25-resets.h>
+#endif
+
 #endif /* __DRIVERS_STM32MP_DT_BINDINGS_H */
diff --git a/core/include/dt-bindings/clock/stm32mp25-clks.h b/core/include/dt-bindings/clock/stm32mp25-clks.h
new file mode 100644
index 000000000..9876ee0dd
--- /dev/null
+++ b/core/include/dt-bindings/clock/stm32mp25-clks.h
@@ -0,0 +1,492 @@
+/* SPDX-License-Identifier: GPL-2.0-only OR BSD-3-Clause */
+/*
+ * Copyright (C) STMicroelectronics 2023 - All Rights Reserved
+ * Author: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
+ */
+
+#ifndef _DT_BINDINGS_STM32MP25_CLKS_H_
+#define _DT_BINDINGS_STM32MP25_CLKS_H_
+
+/* INTERNAL/EXTERNAL OSCILLATORS */
+#define HSI_CK			0
+#define HSE_CK			1
+#define MSI_CK			2
+#define LSI_CK			3
+#define LSE_CK			4
+#define I2S_CK			5
+#define RTC_CK			6
+#define SPDIF_CK_SYMB		7
+
+/* PLL CLOCKS */
+#define PLL1_CK			8
+#define PLL2_CK			9
+#define PLL3_CK			10
+#define PLL4_CK			11
+#define PLL5_CK			12
+#define PLL6_CK			13
+#define PLL7_CK			14
+#define PLL8_CK			15
+
+#define CK_CPU1			16
+
+/* APB DIV CLOCKS */
+#define CK_ICN_APB1		17
+#define CK_ICN_APB2		18
+#define CK_ICN_APB3		19
+#define CK_ICN_APB4		20
+#define CK_ICN_APBDBG		21
+
+/* GLOBAL TIMER */
+#define TIMG1_CK		22
+#define TIMG2_CK		23
+
+/* FLEXGEN CLOCKS */
+#define CK_ICN_HS_MCU		24
+#define CK_ICN_SDMMC		25
+#define CK_ICN_DDR		26
+#define CK_ICN_DISPLAY		27
+#define CK_ICN_HSL		28
+#define CK_ICN_NIC		29
+#define CK_ICN_VID		30
+#define CK_FLEXGEN_07		31
+#define CK_FLEXGEN_08		32
+#define CK_FLEXGEN_09		33
+#define CK_FLEXGEN_10		34
+#define CK_FLEXGEN_11		35
+#define CK_FLEXGEN_12		36
+#define CK_FLEXGEN_13		37
+#define CK_FLEXGEN_14		38
+#define CK_FLEXGEN_15		39
+#define CK_FLEXGEN_16		40
+#define CK_FLEXGEN_17		41
+#define CK_FLEXGEN_18		42
+#define CK_FLEXGEN_19		43
+#define CK_FLEXGEN_20		44
+#define CK_FLEXGEN_21		45
+#define CK_FLEXGEN_22		46
+#define CK_FLEXGEN_23		47
+#define CK_FLEXGEN_24		48
+#define CK_FLEXGEN_25		49
+#define CK_FLEXGEN_26		50
+#define CK_FLEXGEN_27		51
+#define CK_FLEXGEN_28		52
+#define CK_FLEXGEN_29		53
+#define CK_FLEXGEN_30		54
+#define CK_FLEXGEN_31		55
+#define CK_FLEXGEN_32		56
+#define CK_FLEXGEN_33		57
+#define CK_FLEXGEN_34		58
+#define CK_FLEXGEN_35		59
+#define CK_FLEXGEN_36		60
+#define CK_FLEXGEN_37		61
+#define CK_FLEXGEN_38		62
+#define CK_FLEXGEN_39		63
+#define CK_FLEXGEN_40		64
+#define CK_FLEXGEN_41		65
+#define CK_FLEXGEN_42		66
+#define CK_FLEXGEN_43		67
+#define CK_FLEXGEN_44		68
+#define CK_FLEXGEN_45		69
+#define CK_FLEXGEN_46		70
+#define CK_FLEXGEN_47		71
+#define CK_FLEXGEN_48		72
+#define CK_FLEXGEN_49		73
+#define CK_FLEXGEN_50		74
+#define CK_FLEXGEN_51		75
+#define CK_FLEXGEN_52		76
+#define CK_FLEXGEN_53		77
+#define CK_FLEXGEN_54		78
+#define CK_FLEXGEN_55		79
+#define CK_FLEXGEN_56		80
+#define CK_FLEXGEN_57		81
+#define CK_FLEXGEN_58		82
+#define CK_FLEXGEN_59		83
+#define CK_FLEXGEN_60		84
+#define CK_FLEXGEN_61		85
+#define CK_FLEXGEN_62		86
+#define CK_FLEXGEN_63		87
+
+/* LOW SPEED MCU CLOCK */
+#define CK_ICN_LS_MCU		88
+
+#define CK_BUS_STM500		89
+#define CK_BUS_FMC		90
+#define CK_BUS_GPU		91
+#define CK_BUS_ETH1		92
+#define CK_BUS_ETH2		93
+#define CK_BUS_PCIE		94
+#define CK_BUS_DDRPHYC		95
+#define CK_BUS_SYSCPU1		96
+#define CK_BUS_ETHSW		97
+#define CK_BUS_HPDMA1		98
+#define CK_BUS_HPDMA2		99
+#define CK_BUS_HPDMA3		100
+#define CK_BUS_ADC12		101
+#define CK_BUS_ADC3		102
+#define CK_BUS_IPCC1		103
+#define CK_BUS_CCI		104
+#define CK_BUS_CRC		105
+#define CK_BUS_MDF1		106
+#define CK_BUS_OSPIIOM		107
+#define CK_BUS_BKPSRAM		108
+#define CK_BUS_HASH		109
+#define CK_BUS_RNG		110
+#define CK_BUS_CRYP1		111
+#define CK_BUS_CRYP2		112
+#define CK_BUS_SAES		113
+#define CK_BUS_PKA		114
+#define CK_BUS_GPIOA		115
+#define CK_BUS_GPIOB		116
+#define CK_BUS_GPIOC		117
+#define CK_BUS_GPIOD		118
+#define CK_BUS_GPIOE		119
+#define CK_BUS_GPIOF		120
+#define CK_BUS_GPIOG		121
+#define CK_BUS_GPIOH		122
+#define CK_BUS_GPIOI		123
+#define CK_BUS_GPIOJ		124
+#define CK_BUS_GPIOK		125
+#define CK_BUS_LPSRAM1		126
+#define CK_BUS_LPSRAM2		127
+#define CK_BUS_LPSRAM3		128
+#define CK_BUS_GPIOZ		129
+#define CK_BUS_LPDMA		130
+#define CK_BUS_HSEM		131
+#define CK_BUS_IPCC2		132
+#define CK_BUS_RTC		133
+#define CK_BUS_SPI8		134
+#define CK_BUS_LPUART1		135
+#define CK_BUS_I2C8		136
+#define CK_BUS_LPTIM3		137
+#define CK_BUS_LPTIM4		138
+#define CK_BUS_LPTIM5		139
+#define CK_BUS_IWDG5		140
+#define CK_BUS_WWDG2		141
+#define CK_BUS_I3C4		142
+#define CK_BUS_TIM2		143
+#define CK_BUS_TIM3		144
+#define CK_BUS_TIM4		145
+#define CK_BUS_TIM5		146
+#define CK_BUS_TIM6		147
+#define CK_BUS_TIM7		148
+#define CK_BUS_TIM10		149
+#define CK_BUS_TIM11		150
+#define CK_BUS_TIM12		151
+#define CK_BUS_TIM13		152
+#define CK_BUS_TIM14		153
+#define CK_BUS_LPTIM1		154
+#define CK_BUS_LPTIM2		155
+#define CK_BUS_SPI2		156
+#define CK_BUS_SPI3		157
+#define CK_BUS_SPDIFRX		158
+#define CK_BUS_USART2		159
+#define CK_BUS_USART3		160
+#define CK_BUS_UART4		161
+#define CK_BUS_UART5		162
+#define CK_BUS_I2C1		163
+#define CK_BUS_I2C2		164
+#define CK_BUS_I2C3		165
+#define CK_BUS_I2C4		166
+#define CK_BUS_I2C5		167
+#define CK_BUS_I2C6		168
+#define CK_BUS_I2C7		169
+#define CK_BUS_I3C1		170
+#define CK_BUS_I3C2		171
+#define CK_BUS_I3C3		172
+#define CK_BUS_TIM1		173
+#define CK_BUS_TIM8		174
+#define CK_BUS_TIM15		175
+#define CK_BUS_TIM16		176
+#define CK_BUS_TIM17		177
+#define CK_BUS_TIM20		178
+#define CK_BUS_SAI1		179
+#define CK_BUS_SAI2		180
+#define CK_BUS_SAI3		181
+#define CK_BUS_SAI4		182
+#define CK_BUS_USART1		183
+#define CK_BUS_USART6		184
+#define CK_BUS_UART7		185
+#define CK_BUS_UART8		186
+#define CK_BUS_UART9		187
+#define CK_BUS_FDCAN		188
+#define CK_BUS_SPI1		189
+#define CK_BUS_SPI4		190
+#define CK_BUS_SPI5		191
+#define CK_BUS_SPI6		192
+#define CK_BUS_SPI7		193
+#define CK_BUS_BSEC		194
+#define CK_BUS_IWDG1		195
+#define CK_BUS_IWDG2		196
+#define CK_BUS_IWDG3		197
+#define CK_BUS_IWDG4		198
+#define CK_BUS_WWDG1		199
+#define CK_BUS_VREF		200
+#define CK_BUS_DTS		201
+#define CK_BUS_SERC		202
+#define CK_BUS_HDP		203
+#define CK_BUS_IS2M		204
+#define CK_BUS_DSI		205
+#define CK_BUS_LTDC		206
+#define CK_BUS_CSI		207
+#define CK_BUS_DCMIPP		208
+#define CK_BUS_DDRC		209
+#define CK_BUS_DDRCFG		210
+#define CK_BUS_GICV2M		211
+#define CK_BUS_USBTC		212
+#define CK_BUS_USB3PCIEPHY	214
+#define CK_BUS_STGEN		215
+#define CK_BUS_VDEC		216
+#define CK_BUS_VENC		217
+#define CK_SYSDBG		218
+#define CK_KER_TIM2		219
+#define CK_KER_TIM3		220
+#define CK_KER_TIM4		221
+#define CK_KER_TIM5		222
+#define CK_KER_TIM6		223
+#define CK_KER_TIM7		224
+#define CK_KER_TIM10		225
+#define CK_KER_TIM11		226
+#define CK_KER_TIM12		227
+#define CK_KER_TIM13		228
+#define CK_KER_TIM14		229
+#define CK_KER_TIM1		230
+#define CK_KER_TIM8		231
+#define CK_KER_TIM15		232
+#define CK_KER_TIM16		233
+#define CK_KER_TIM17		234
+#define CK_KER_TIM20		235
+#define CK_BUS_SYSRAM		236
+#define CK_BUS_VDERAM		237
+#define CK_BUS_RETRAM		238
+#define CK_BUS_OSPI1		239
+#define CK_BUS_OSPI2		240
+#define CK_BUS_OTFD1		241
+#define CK_BUS_OTFD2		242
+#define CK_BUS_SRAM1		243
+#define CK_BUS_SRAM2		244
+#define CK_BUS_SDMMC1		245
+#define CK_BUS_SDMMC2		246
+#define CK_BUS_SDMMC3		247
+#define CK_BUS_DDR		248
+#define CK_BUS_RISAF4		249
+#define CK_BUS_USB2OHCI		250
+#define CK_BUS_USB2EHCI		251
+#define CK_BUS_USB3DR		252
+#define CK_KER_LPTIM1		253
+#define CK_KER_LPTIM2		254
+#define CK_KER_USART2		255
+#define CK_KER_UART4		256
+#define CK_KER_USART3		257
+#define CK_KER_UART5		258
+#define CK_KER_SPI2		259
+#define CK_KER_SPI3		260
+#define CK_KER_SPDIFRX		261
+#define CK_KER_I2C1		262
+#define CK_KER_I2C2		263
+#define CK_KER_I3C1		264
+#define CK_KER_I3C2		265
+#define CK_KER_I2C3		266
+#define CK_KER_I2C5		267
+#define CK_KER_I3C3		268
+#define CK_KER_I2C4		269
+#define CK_KER_I2C6		270
+#define CK_KER_I2C7		271
+#define CK_KER_SPI1		272
+#define CK_KER_SPI4		273
+#define CK_KER_SPI5		274
+#define CK_KER_SPI6		275
+#define CK_KER_SPI7		276
+#define CK_KER_USART1		277
+#define CK_KER_USART6		278
+#define CK_KER_UART7		279
+#define CK_KER_UART8		280
+#define CK_KER_UART9		281
+#define CK_KER_MDF1		282
+#define CK_KER_SAI1		283
+#define CK_KER_SAI2		284
+#define CK_KER_SAI3		285
+#define CK_KER_SAI4		286
+#define CK_KER_FDCAN		287
+#define CK_KER_DSIBLANE		288
+#define CK_KER_DSIPHY		289
+#define CK_KER_CSI		290
+#define CK_KER_CSITXESC		291
+#define CK_KER_CSIPHY		292
+#define CK_KER_LVDSPHY		293
+#define CK_KER_STGEN		294
+#define CK_KER_USB3PCIEPHY	295
+#define CK_KER_USB2PHY2EN	296
+#define CK_KER_I3C4		297
+#define CK_KER_SPI8		298
+#define CK_KER_I2C8		299
+#define CK_KER_LPUART1		300
+#define CK_KER_LPTIM3		301
+#define CK_KER_LPTIM4		302
+#define CK_KER_LPTIM5		303
+#define CK_KER_TSDBG		304
+#define CK_KER_TPIU		305
+#define CK_BUS_ETR		306
+#define CK_BUS_SYSATB		307
+#define CK_KER_ADC12		308
+#define CK_KER_ADC3		309
+#define CK_KER_OSPI1		310
+#define CK_KER_OSPI2		311
+#define CK_KER_FMC		312
+#define CK_KER_SDMMC1		313
+#define CK_KER_SDMMC2		314
+#define CK_KER_SDMMC3		315
+#define CK_KER_ETH1		316
+#define CK_KER_ETH2		317
+#define CK_KER_ETH1PTP		318
+#define CK_KER_ETH2PTP		319
+#define CK_KER_USB2PHY1		320
+#define CK_KER_USB2PHY2		321
+#define CK_KER_ETHSW		322
+#define CK_KER_ETHSWREF		323
+#define CK_MCO1			324
+#define CK_MCO2			325
+#define CK_KER_DTS		326
+#define CK_ETH1_RX		327
+#define CK_ETH1_TX		328
+#define CK_ETH1_MAC		329
+#define CK_ETH2_RX		330
+#define CK_ETH2_TX		331
+#define CK_ETH2_MAC		332
+#define CK_ETH1_STP		333
+#define CK_ETH2_STP		334
+#define CK_KER_USBTC		335
+#define CK_BUS_ADF1		336
+#define CK_KER_ADF1		337
+#define CK_BUS_LVDS		338
+#define CK_KER_LTDC		339
+#define CK_KER_GPU		340
+#define CK_BUS_ETHSWACMCFG	341
+#define CK_BUS_ETHSWACMMSG	342
+#define HSE_DIV2_CK		343
+
+#define STM32MP25_LAST_CLK	344
+
+#define CK_SCMI_ICN_HS_MCU	0
+#define CK_SCMI_ICN_SDMMC	1
+#define CK_SCMI_ICN_DDR		2
+#define CK_SCMI_ICN_DISPLAY	3
+#define CK_SCMI_ICN_HSL		4
+#define CK_SCMI_ICN_NIC		5
+#define CK_SCMI_ICN_VID		6
+#define CK_SCMI_FLEXGEN_07	7
+#define CK_SCMI_FLEXGEN_08	8
+#define CK_SCMI_FLEXGEN_09	9
+#define CK_SCMI_FLEXGEN_10	10
+#define CK_SCMI_FLEXGEN_11	11
+#define CK_SCMI_FLEXGEN_12	12
+#define CK_SCMI_FLEXGEN_13	13
+#define CK_SCMI_FLEXGEN_14	14
+#define CK_SCMI_FLEXGEN_15	15
+#define CK_SCMI_FLEXGEN_16	16
+#define CK_SCMI_FLEXGEN_17	17
+#define CK_SCMI_FLEXGEN_18	18
+#define CK_SCMI_FLEXGEN_19	19
+#define CK_SCMI_FLEXGEN_20	20
+#define CK_SCMI_FLEXGEN_21	21
+#define CK_SCMI_FLEXGEN_22	22
+#define CK_SCMI_FLEXGEN_23	23
+#define CK_SCMI_FLEXGEN_24	24
+#define CK_SCMI_FLEXGEN_25	25
+#define CK_SCMI_FLEXGEN_26	26
+#define CK_SCMI_FLEXGEN_27	27
+#define CK_SCMI_FLEXGEN_28	28
+#define CK_SCMI_FLEXGEN_29	29
+#define CK_SCMI_FLEXGEN_30	30
+#define CK_SCMI_FLEXGEN_31	31
+#define CK_SCMI_FLEXGEN_32	32
+#define CK_SCMI_FLEXGEN_33	33
+#define CK_SCMI_FLEXGEN_34	34
+#define CK_SCMI_FLEXGEN_35	35
+#define CK_SCMI_FLEXGEN_36	36
+#define CK_SCMI_FLEXGEN_37	37
+#define CK_SCMI_FLEXGEN_38	38
+#define CK_SCMI_FLEXGEN_39	39
+#define CK_SCMI_FLEXGEN_40	40
+#define CK_SCMI_FLEXGEN_41	41
+#define CK_SCMI_FLEXGEN_42	42
+#define CK_SCMI_FLEXGEN_43	43
+#define CK_SCMI_FLEXGEN_44	44
+#define CK_SCMI_FLEXGEN_45	45
+#define CK_SCMI_FLEXGEN_46	46
+#define CK_SCMI_FLEXGEN_47	47
+#define CK_SCMI_FLEXGEN_48	48
+#define CK_SCMI_FLEXGEN_49	49
+#define CK_SCMI_FLEXGEN_50	50
+#define CK_SCMI_FLEXGEN_51	51
+#define CK_SCMI_FLEXGEN_52	52
+#define CK_SCMI_FLEXGEN_53	53
+#define CK_SCMI_FLEXGEN_54	54
+#define CK_SCMI_FLEXGEN_55	55
+#define CK_SCMI_FLEXGEN_56	56
+#define CK_SCMI_FLEXGEN_57	57
+#define CK_SCMI_FLEXGEN_58	58
+#define CK_SCMI_FLEXGEN_59	59
+#define CK_SCMI_FLEXGEN_60	60
+#define CK_SCMI_FLEXGEN_61	61
+#define CK_SCMI_FLEXGEN_62	62
+#define CK_SCMI_FLEXGEN_63	63
+#define CK_SCMI_ICN_LS_MCU	64
+#define CK_SCMI_HSE		65
+#define CK_SCMI_LSE		66
+#define CK_SCMI_HSI		67
+#define CK_SCMI_LSI		68
+#define CK_SCMI_MSI		69
+#define CK_SCMI_HSE_DIV2	70
+#define CK_SCMI_CPU1		71
+#define CK_SCMI_SYSCPU1		72
+#define CK_SCMI_PLL2		73
+#define CK_SCMI_PLL3		74
+#define CK_SCMI_RTC		75
+#define CK_SCMI_RTCCK		76
+#define CK_SCMI_ICN_APB1	77
+#define CK_SCMI_ICN_APB2	78
+#define CK_SCMI_ICN_APB3	79
+#define CK_SCMI_ICN_APB4	80
+#define CK_SCMI_ICN_APBDBG	81
+#define CK_SCMI_TIMG1		82
+#define CK_SCMI_TIMG2		83
+#define CK_SCMI_BKPSRAM		84
+#define CK_SCMI_BSEC		85
+#define CK_SCMI_ETR		87
+#define CK_SCMI_FMC		88
+#define CK_SCMI_GPIOA		89
+#define CK_SCMI_GPIOB		90
+#define CK_SCMI_GPIOC		91
+#define CK_SCMI_GPIOD		92
+#define CK_SCMI_GPIOE		93
+#define CK_SCMI_GPIOF		94
+#define CK_SCMI_GPIOG		95
+#define CK_SCMI_GPIOH		96
+#define CK_SCMI_GPIOI		97
+#define CK_SCMI_GPIOJ		98
+#define CK_SCMI_GPIOK		99
+#define CK_SCMI_GPIOZ		100
+#define CK_SCMI_HPDMA1		101
+#define CK_SCMI_HPDMA2		102
+#define CK_SCMI_HPDMA3		103
+#define CK_SCMI_HSEM		104
+#define CK_SCMI_IPCC1		105
+#define CK_SCMI_IPCC2		106
+#define CK_SCMI_LPDMA		107
+#define CK_SCMI_RETRAM		108
+#define CK_SCMI_SRAM1		109
+#define CK_SCMI_SRAM2		110
+#define CK_SCMI_LPSRAM1		111
+#define CK_SCMI_LPSRAM2		112
+#define CK_SCMI_LPSRAM3		113
+#define CK_SCMI_VDERAM		114
+#define CK_SCMI_SYSRAM		115
+#define CK_SCMI_OSPI1		116
+#define CK_SCMI_OSPI2		117
+#define CK_SCMI_TPIU		118
+#define CK_SCMI_SYSDBG		119
+#define CK_SCMI_SYSATB		120
+#define CK_SCMI_TSDBG		121
+#define CK_SCMI_STM500		122
+
+#endif /* _DT_BINDINGS_STM32MP25_CLKS_H_ */
diff --git a/core/include/dt-bindings/clock/stm32mp25-clksrc.h b/core/include/dt-bindings/clock/stm32mp25-clksrc.h
new file mode 100644
index 000000000..24ba580ab
--- /dev/null
+++ b/core/include/dt-bindings/clock/stm32mp25-clksrc.h
@@ -0,0 +1,226 @@
+/* SPDX-License-Identifier: GPL-2.0+ OR BSD-3-Clause */
+/*
+ * Copyright (C) 2017-2020, STMicroelectronics - All Rights Reserved
+ */
+
+#ifndef _DT_BINDINGS_CLOCK_STM32MP2_CLKSRC_H_
+#define _DT_BINDINGS_CLOCK_STM32MP2_CLKSRC_H_
+
+#define CMD_DIV		0
+#define CMD_MUX		1
+#define CMD_CLK		2
+#define CMD_FLEXGEN	3
+
+#define CMD_ADDR_BIT	0x80000000
+
+#define CMD_SHIFT	26
+#define CMD_MASK	0xFC000000
+#define CMD_DATA_MASK	0x03FFFFFF
+
+#define DIV_ID_SHIFT	8
+#define DIV_ID_MASK	0x0000FF00
+
+#define DIV_DIVN_SHIFT	0
+#define DIV_DIVN_MASK	0x000000FF
+
+#define MUX_ID_SHIFT	4
+#define MUX_ID_MASK	0x00000FF0
+
+#define MUX_SEL_SHIFT	0
+#define MUX_SEL_MASK	0x0000000F
+
+/* CLK define */
+#define CLK_ON_MASK	BIT(21)
+#define CLK_ON_SHIFT	21
+
+#define CLK_ID_MASK	GENMASK_32(20, 12)
+#define CLK_ID_SHIFT	12
+
+#define CLK_NO_DIV_MASK	0x0000080
+#define CLK_DIV_MASK	GENMASK_32(10, 5)
+#define CLK_DIV_SHIFT	5
+
+#define CLK_NO_SEL_MASK	0x00000010
+#define CLK_SEL_MASK	GENMASK_32(3, 0)
+#define CLK_SEL_SHIFT	0
+
+#define CLK_CFG(clk_id, sel, div, state)	((CMD_CLK << CMD_SHIFT) |\
+						 ((state) << CLK_ON_SHIFT) |\
+						 ((clk_id) << CLK_ID_SHIFT) |\
+						 ((div) << CLK_DIV_SHIFT) |\
+						 ((sel) << CLK_SEL_SHIFT))
+
+#define CLK_OFF		0
+#define CLK_ON		1
+#define CLK_NODIV	0x00000040
+#define CLK_NOMUX	0x00000010
+
+/* Flexgen define */
+#define FLEX_ID_SHIFT	13
+#define FLEX_SEL_SHIFT	9
+#define FLEX_PDIV_SHIFT	6
+#define FLEX_FDIV_SHIFT	0
+
+#define FLEX_ID_MASK	GENMASK_32(18, 13)
+#define FLEX_SEL_MASK	GENMASK_32(12, 9)
+#define FLEX_PDIV_MASK	GENMASK_32(8, 6)
+#define FLEX_FDIV_MASK	GENMASK_32(5, 0)
+
+#define DIV_CFG(div_id, div)	((CMD_DIV << CMD_SHIFT) |\
+				 ((div_id) << DIV_ID_SHIFT |\
+				 (div)))
+
+#define MUX_CFG(mux_id, sel)	((CMD_MUX << CMD_SHIFT) |\
+				 ((mux_id) << MUX_ID_SHIFT |\
+				 (sel)))
+
+#define CLK_ADDR_SHIFT		16
+#define CLK_ADDR_MASK		0x7FFF0000
+#define CLK_ADDR_VAL_MASK	0xFFFF
+
+#define DIV_LSMCU	0
+#define DIV_APB1	1
+#define DIV_APB2	2
+#define DIV_APB3	3
+#define DIV_APB4	4
+#define DIV_APBDBG	5
+#define DIV_RTC		6
+#define DIV_NB		7
+
+#define MUX_MUXSEL0	0
+#define MUX_MUXSEL1	1
+#define MUX_MUXSEL2	2
+#define MUX_MUXSEL3	3
+#define MUX_MUXSEL4	4
+#define MUX_MUXSEL5	5
+#define MUX_MUXSEL6	6
+#define MUX_MUXSEL7	7
+#define MUX_XBARSEL	8
+#define MUX_RTC		9
+#define MUX_MCO1	10
+#define MUX_MCO2	11
+#define MUX_ADC12	12
+#define MUX_ADC3	13
+#define MUX_USB2PHY1	14
+#define MUX_USB2PHY2	15
+#define MUX_USB3PCIEPHY	16
+#define MUX_DSIBLANE	17
+#define MUX_DSIPHY	18
+#define MUX_LVDSPHY	19
+#define MUX_DTS		20
+#define MUX_CPU1	21
+#define MUX_D3PER	22
+#define MUX_NB		23
+
+#define MUXSEL_HSI		0
+#define MUXSEL_HSE		1
+#define MUXSEL_MSI		2
+
+/* KERNEL source clocks */
+#define MUX_RTC_DISABLED	0x0
+#define MUX_RTC_LSE		0x1
+#define MUX_RTC_LSI		0x2
+#define MUX_RTC_HSE		0x3
+
+#define MUX_MCO1_FLEX61		0x0
+#define MUX_MCO1_OBSER0		0x1
+
+#define MUX_MCO2_FLEX62		0x0
+#define MUX_MCO2_OBSER1		0x1
+
+#define MUX_ADC12_FLEX46	0x0
+#define MUX_ADC12_LSMCU		0x1
+
+#define MUX_ADC3_FLEX47		0x0
+#define MUX_ADC3_LSMCU		0x1
+#define MUX_ADC3_FLEX46		0x2
+
+#define MUX_USB2PHY1_FLEX57	0x0
+#define MUX_USB2PHY1_HSE	0x1
+
+#define MUX_USB2PHY2_FLEX58	0x0
+#define MUX_USB2PHY2_HSE	0x1
+
+#define MUX_USB3PCIEPHY_FLEX34	0x0
+#define MUX_USB3PCIEPHY_HSE	0x1
+
+#define MUX_DSIBLANE_DSIPHY	0x0
+#define MUX_DSIBLANE_FLEX27	0x1
+
+#define MUX_DSIPHY_FLEX28	0x0
+#define MUX_DSIPHY_HSE		0x1
+
+#define MUX_LVDSPHY_FLEX32	0x0
+#define MUX_LVDSPHY_HSE		0x1
+
+#define MUX_DTS_HSI		0x0
+#define MUX_DTS_HSE		0x1
+#define MUX_DTS_MSI		0x2
+
+#define MUX_D3PER_MSI		0x0
+#define MUX_D3PER_LSI		0x1
+#define MUX_D3PER_LSE		0x2
+
+/* PLLs source clocks */
+#define PLL_SRC_HSI		0x0
+#define PLL_SRC_HSE		0x1
+#define PLL_SRC_MSI		0x2
+#define PLL_SRC_DISABLED	0x3
+
+/* XBAR source clocks */
+#define XBAR_SRC_PLL4		0x0
+#define XBAR_SRC_PLL5		0x1
+#define XBAR_SRC_PLL6		0x2
+#define XBAR_SRC_PLL7		0x3
+#define XBAR_SRC_PLL8		0x4
+#define XBAR_SRC_HSI		0x5
+#define XBAR_SRC_HSE		0x6
+#define XBAR_SRC_MSI		0x7
+#define XBAR_SRC_HSI_KER	0x8
+#define XBAR_SRC_HSE_KER	0x9
+#define XBAR_SRC_MSI_KER	0xA
+#define XBAR_SRC_SPDIF_SYMB	0xB
+#define XBAR_SRC_I2S		0xC
+#define XBAR_SRC_LSI		0xD
+#define XBAR_SRC_LSE		0xE
+
+/*
+ * Configure a XBAR channel with its clock source
+ * channel_nb: XBAR channel number from 0 to 63
+ * channel_src: one of the 15 previous XBAR source clocks defines
+ * channel_prediv: value of the PREDIV in channel RCC_PREDIVxCFGR register
+ *		   can be either 1, 2, 4 or 1024
+ * channel_findiv: value of the FINDIV in channel RCC_FINDIVxCFGR register
+ *		   from 1 to 64
+ */
+
+#define FLEXGEN_CFG(ch, sel, pdiv, fdiv)	((CMD_FLEXGEN << CMD_SHIFT) |\
+						((ch) << FLEX_ID_SHIFT) |\
+						((sel) << FLEX_SEL_SHIFT) |\
+						((pdiv) << FLEX_PDIV_SHIFT) |\
+						((fdiv) << FLEX_FDIV_SHIFT))
+
+/* Register addresses of MCO1 & MCO2 */
+#define MCO1			0x494
+#define MCO2			0x498
+
+#define MCO_OFF			0
+#define MCO_ON			1
+#define MCO_STATUS_SHIFT	8
+
+#define MCO_CFG(addr, sel, status)	(CMD_ADDR_BIT |\
+					((addr) << CLK_ADDR_SHIFT) |\
+					((status) << MCO_STATUS_SHIFT) |\
+					(sel))
+
+/* define for st,pll /csg */
+#define SSCG_MODE_CENTER_SPREAD	0
+#define SSCG_MODE_DOWN_SPREAD	1
+
+/* define for st,drive */
+#define LSEDRV_LOWEST		0
+#define LSEDRV_MEDIUM_LOW	2
+#define LSEDRV_MEDIUM_HIGH	1
+#define LSEDRV_HIGHEST		3
+
+#endif /* _DT_BINDINGS_CLOCK_STM32MP2_CLKSRC_H_ */
diff --git a/core/include/dt-bindings/reset/stm32mp25-resets.h b/core/include/dt-bindings/reset/stm32mp25-resets.h
new file mode 100644
index 000000000..7868f1425
--- /dev/null
+++ b/core/include/dt-bindings/reset/stm32mp25-resets.h
@@ -0,0 +1,167 @@
+/* SPDX-License-Identifier: GPL-2.0-only OR BSD-3-Clause */
+/*
+ * Copyright (C) STMicroelectronics 2023 - All Rights Reserved
+ * Author(s): Gabriel Fernandez <gabriel.fernandez@foss.st.com>
+ */
+
+#ifndef _DT_BINDINGS_STM32MP25_RESET_H_
+#define _DT_BINDINGS_STM32MP25_RESET_H_
+
+#define TIM1_R		0
+#define TIM2_R		1
+#define TIM3_R		2
+#define TIM4_R		3
+#define TIM5_R		4
+#define TIM6_R		5
+#define TIM7_R		6
+#define TIM8_R		7
+#define TIM10_R		8
+#define TIM11_R		9
+#define TIM12_R		10
+#define TIM13_R		11
+#define TIM14_R		12
+#define TIM15_R		13
+#define TIM16_R		14
+#define TIM17_R		15
+#define TIM20_R		16
+#define LPTIM1_R	17
+#define LPTIM2_R	18
+#define LPTIM3_R	19
+#define LPTIM4_R	20
+#define LPTIM5_R	21
+#define SPI1_R		22
+#define SPI2_R		23
+#define SPI3_R		24
+#define SPI4_R		25
+#define SPI5_R		26
+#define SPI6_R		27
+#define SPI7_R		28
+#define SPI8_R		29
+#define SPDIFRX_R	30
+#define USART1_R	31
+#define USART2_R	32
+#define USART3_R	33
+#define UART4_R		34
+#define UART5_R		35
+#define USART6_R	36
+#define UART7_R		37
+#define UART8_R		38
+#define UART9_R		39
+#define LPUART1_R	40
+#define IS2M_R		41
+#define I2C1_R		42
+#define I2C2_R		43
+#define I2C3_R		44
+#define I2C4_R		45
+#define I2C5_R		46
+#define I2C6_R		47
+#define I2C7_R		48
+#define I2C8_R		49
+#define SAI1_R		50
+#define SAI2_R		51
+#define SAI3_R		52
+#define SAI4_R		53
+#define MDF1_R		54
+#define MDF2_R		55
+#define FDCAN_R		56
+#define HDP_R		57
+#define ADC12_R		58
+#define ADC3_R		59
+#define ETH1_R		60
+#define ETH2_R		61
+#define USB2_R		62
+#define USB2PHY1_R	63
+#define USB2PHY2_R	64
+#define USB3DR_R	65
+#define USB3PCIEPHY_R	66
+#define USBTC_R		67
+#define ETHSW_R		68
+#define SDMMC1_R	69
+#define SDMMC1DLL_R	70
+#define SDMMC2_R	71
+#define SDMMC2DLL_R	72
+#define SDMMC3_R	73
+#define SDMMC3DLL_R	74
+#define GPU_R		75
+#define LTDC_R		76
+#define DSI_R		77
+#define LVDS_R		78
+#define CSI_R		79
+#define DCMIPP_R	80
+#define CCI_R		81
+#define VDEC_R		82
+#define VENC_R		83
+#define WWDG1_R		84
+#define WWDG2_R		85
+#define VREF_R		86
+#define DTS_R		87
+#define CRC_R		88
+#define SERC_R		89
+#define OSPIIOM_R	90
+#define I3C1_R		91
+#define I3C2_R		92
+#define I3C3_R		93
+#define I3C4_R		94
+#define IWDG2_KER_R	95
+#define IWDG4_KER_R	96
+#define RNG_R		97
+#define PKA_R		98
+#define SAES_R		99
+#define HASH_R		100
+#define CRYP1_R		101
+#define CRYP2_R		102
+#define PCIE_R		103
+#define OSPI1_R		104
+#define OSPI1DLL_R	105
+#define OSPI2_R		106
+#define OSPI2DLL_R	107
+#define FMC_R		108
+#define DBG_R		109
+#define GPIOA_R		110
+#define GPIOB_R		111
+#define GPIOC_R		112
+#define GPIOD_R		113
+#define GPIOE_R		114
+#define GPIOF_R		115
+#define GPIOG_R		116
+#define GPIOH_R		117
+#define GPIOI_R		118
+#define GPIOJ_R		119
+#define GPIOK_R		120
+#define GPIOZ_R		121
+#define HPDMA1_R	122
+#define HPDMA2_R	123
+#define HPDMA3_R	124
+#define LPDMA_R		125
+#define HSEM_R		126
+#define IPCC1_R		127
+#define IPCC2_R		128
+#define C2_HOLDBOOT_R	129
+#define C1_HOLDBOOT_R	130
+#define C1_R		131
+#define C1P1POR_R	132
+#define C1P1_R		133
+#define C2_R		134
+#define C3_R		135
+#define SYS_R		136
+#define VSW_R		137
+#define C1MS_R		138
+#define DDRCP_R		139
+#define DDRCAPB_R	140
+#define DDRPHYCAPB_R	141
+#define DDRCFG_R	142
+#define DDR_R		143
+
+#define STM32MP25_LAST_RESET	144
+
+#define RST_SCMI_C1_R		0
+#define RST_SCMI_C2_R		1
+#define RST_SCMI_C1_HOLDBOOT_R	2
+#define RST_SCMI_C2_HOLDBOOT_R	3
+#define RST_SCMI_FMC		4
+#define RST_SCMI_OSPI1		5
+#define RST_SCMI_OSPI1DLL	6
+#define RST_SCMI_OSPI2		7
+#define RST_SCMI_OSPI2DLL	8
+
+#endif /* _DT_BINDINGS_STM32MP25_RESET_H_ */
diff --git a/documentation/devicetree/bindings/clock/st,stm32mp25-rcc.yaml b/documentation/devicetree/bindings/clock/st,stm32mp25-rcc.yaml
new file mode 100644
index 000000000..3d196e36e
--- /dev/null
+++ b/documentation/devicetree/bindings/clock/st,stm32mp25-rcc.yaml
@@ -0,0 +1,448 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/clock/st,stm32mp25-rcc.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: STM32MP25 Reset Clock Controller
+
+maintainers:
+  - Gabriel Fernandez <gabriel.fernandez@foss.st.com>
+
+description: |
+  The RCC hardware block is both a reset and a clock controller.
+  RCC makes also power management (resume/supend).
+
+  See also::
+    include/dt-bindings/clock/st,stm32mp25-rcc.h
+    include/dt-bindings/reset/st,stm32mp25-rcc.h
+
+properties:
+  compatible:
+    enum:
+      - st,stm32mp25-rcc
+
+  reg:
+    maxItems: 1
+
+  '#clock-cells':
+    const: 1
+
+  '#reset-cells':
+    const: 1
+
+  clocks:
+    items:
+      - description: CK_SCMI_HSE High Speed External oscillator (8 to 48 MHz)
+      - description: CK_SCMI_HSI High Speed Internal oscillator (~ 64 MHz)
+      - description: CK_SCMI_MSI Low Power Internal oscillator (~ 4 MHz or ~ 16 MHz)
+      - description: CK_SCMI_LSE Low Speed External oscillator (32 KHz)
+      - description: CK_SCMI_LSI Low Speed Internal oscillator (~ 32 KHz)
+
+  clock-names:
+    items:
+      - const: hse
+      - const: hsi
+      - const: msi
+      - const: lse
+      - const: lsi
+
+  interrupts:
+    description: contains the general interrupt line (for HSI calibration).
+    maxItems: 1
+
+  st,c1msrd:
+    description: |
+        - It defines the minimum duration (in ms) of the low level pulse on
+        NRSTC1MS pad when C1MSRST is high (see ref manual for details)
+    minItems: 1
+
+  st,flexgen:
+    description: |
+        - Used to configure the flexgen clock selection.
+        The property is a list of flex clock source identifiers defined
+        by macros as defined by header file dt-bindings/clock/stm32mp25-clksrc.h:
+          - FLEXGEN_CFG(chanel, XBAR_SRC_<PARENT-CLOCK>, pre_div, final_div)
+            - pre_div
+                0x0: not divided
+                0x1: division by 2
+                0x3: division by 4
+                0x3ff: division by 1024
+            - final_div
+                0x0: not divided
+                0x1: division by 2
+                0x2: division by 3
+                ...
+                0x3f: division by 64
+        - Property can be used to configure the clock distribution tree:
+            st,flexgen = <
+                    FLEXGEN_CFG(0, XBAR_SRC_PLL4, 0, 2)
+                    FLEXGEN_CFG(2, XBAR_SRC_PLL4, 0, 1)
+                    FLEXGEN_CFG(3, XBAR_SRC_PLL4, 0, 2)
+                    FLEXGEN_CFG(4, XBAR_SRC_PLL4, 0, 3)
+                    FLEXGEN_CFG(5, XBAR_SRC_PLL4, 0, 2)
+                    FLEXGEN_CFG(6, XBAR_SRC_PLL4, 0, 1)
+                    FLEXGEN_CFG(7, XBAR_SRC_PLL4, 0, 11)
+                    ...
+                    FLEXGEN_CFG(59, XBAR_SRC_PLL4, 0, 1)
+                    FLEXGEN_CFG(60, XBAR_SRC_PLL6, 0, 3)
+                    FLEXGEN_CFG(61, XBAR_SRC_PLL4, 0, 7)
+                    FLEXGEN_CFG(62, XBAR_SRC_PLL4, 0, 7)
+                    FLEXGEN_CFG(63, XBAR_SRC_PLL4, 0, 2)
+            >;
+    $ref: "/schemas/types.yaml#/definitions/uint32-array"
+
+  st,busclk:
+    description: |
+      - The property is a list of dividers defined by macros DIV(DIV_<DIV-CLOCK>,
+        value) as defined by header file dt-bindings/clock/stm32mp25-clksrc.h.
+      - For the STM32MP25 family there are 8 dividers values expected:
+        LSMCU APB1 APB2 APB3 APB4 DIV_APBDBG
+      - Each divider value uses the DIV coding defined in RCC associated register
+        RCC_xxxDIVR. In most cases, it is:
+          0x0: not divided
+          0x1: division by 2
+          0x2: division by 4
+          0x3: division by 8
+          ...
+      - Property can be used to configure the clock main dividers value:
+          st,busclk = <
+            DIV(DIV_LSMCU, 1)
+            DIV(DIV_APB1, 0)
+            DIV(DIV_APB2, 0)
+            DIV(DIV_APB3, 0)
+            DIV(DIV_APB4, 0)
+            DIV(DIV_APBDBG, 0)
+          >;
+    $ref: "/schemas/types.yaml#/definitions/uint32-array"
+
+  st,kerclk:
+    description: |
+        - used to configure the peripherals kernel clock selection.
+        The property is a list of peripheral kernel clock source identifiers defined
+        by macros as defined by header file dt-bindings/clock/stm32mp25-clksrc.h:
+          - DIV(DIV_<DIV-CLOCK>, value)
+          - MUX_CFG(MUX_<KERNEL-CLOCK>, MUX<KERNEL-CLOCK>_<PARENT-CLOCK>
+          - MCO_CFG(MCO_CLOCK, MUX_<MCO-CLOCK>_<PARENT-CLOCK>, ON / OFF)
+        - st,kerclk may not list all the kernel clocks and has no ordering requirements.
+        - Property can be used to configure the clock distribution tree:
+            st,kerclk = <
+                    DIV_CFG(DIV_RTC, 23)
+                    MUX_CFG(MUX_ADC12, MUX_ADC12_FLEX46)
+                    MUX_CFG(MUX_ADC3, MUX_ADC3_FLEX47)
+                    MUX_CFG(MUX_USB2PHY1, MUX_USB2PHY1_FLEX57)
+                    MUX_CFG(MUX_USB2PHY2, MUX_USB2PHY2_FLEX58)
+                    MUX_CFG(MUX_USB3PCIEPHY, MUX_USB3PCIEPHY_FLEX34)
+                    MUX_CFG(MUX_DSIPHY, MUX_DSIPHY_FLEX28)
+                    MUX_CFG(MUX_DSIBLANE, MUX_DSIBLANE_DSIPHY)
+                    MUX_CFG(MUX_LVDSPHY, MUX_LVDSPHY_FLEX32)
+                    MUX_CFG(MUX_DTS, MUX_DTS_HSE)
+                    MUX_CFG(MUX_RTC, MUX_RTC_HSE)
+                    MUX_CFG(MUX_D3PER, MUX_D3PER_LSE)
+                    MCO_CFG(MCO2, MUX_MCO2_FLEX62, MCO_OFF)
+            >;
+    $ref: "/schemas/types.yaml#/definitions/uint32-array"
+
+patternProperties:
+  '^st,pll-[1-8]$':
+    description:
+      Each PLL children node for PLL1 to PLL8 (see ref manual for details).
+      PLL3..PLL8 are off when their associated nodes are absent or deactivated.
+    type: object
+    properties:
+      st,pll:
+        description:
+          Phandle of the default pll configuration.
+          A pll could have several configuration (5 max) and should be described in a subnode
+          just below.
+        $ref: "/schemas/types.yaml#/definitions/phandle-array"
+    patternProperties:
+      '^pll[1-8]_cfg[0-9]*$':
+        type: object
+        properties:
+          cfg:
+             $ref: "/schemas/types.yaml#/definitions/uint32-array"
+             description:
+              The parameters for PLL frequency configuration in the following order
+              <FBDIV FREFDIV POSTDIV1 POSTDIV2>
+              FREF * (FBDIV / FREFDIV) / POSTDIV1 / POSTDIV2
+              POSTDIVx
+                0x0: not applicable
+                0x1: division by 1
+                ...
+                0x7: division by 7
+              (see ref manual for details)
+            minItems: 1
+          src:
+            description:
+              Clock source configuration values are defined by macros MUX_CFG(MUX_<NAME>_<SOURCE>)
+              from dt-bindings/clock/stm32mp25-clksrc.h.
+            minItems: 1
+          frac:
+            description:
+              Fractional part of the multiplication factor
+              (optional, PLL is in integer mode when absent).
+            minItems: 1
+          csg:
+            description: |
+              Clock Spreading Generator (optional) with parameters in the
+              following order: MOD_PER INC_STEP SSCG_MODE.
+              MOD_PER: Modulation Period Adjustment
+              INC_STEP: Modulation Depth Adjustment
+              SSCG_MODE: Spread spectrum clock generator mode, with associated
+              defines from stm32mp25-clksrc.h:
+              - SSCG_MODE_CENTER_SPREAD = 0
+              - SSCG_MODE_DOWN_SPREAD = 1
+            minItems: 1
+    additionalProperties: false
+
+  '^st,clk_opp':
+    description: |
+      This node will contain all operating point configurations. The OPP
+      configuration will be applied at the run-time and will override the clock tree
+      configuration.
+      Each node will contain a list of operating point configurations (cfg_1, cfg_2...)
+    type: object
+    patternProperties:
+      '^st,cpu1_[a-z]*$':
+        type: object
+        description: |
+          three kind of operating points: st,ck_cpu1.
+        patternProperties:
+          'cfg_[0-9]*$':
+            type: object
+            description: list of operating point configuration.
+            properties:
+              hz:
+                description: Operating point frequency in hertz
+                minItems: 1
+              st,pll:
+                description: Phandle of the pll configuration (See description of 'pllx_cfgx' node)
+                $ref: "/schemas/types.yaml#/definitions/phandle-array"
+                minItems: 1
+            required:
+              - hz
+              - st,pll
+            additionalProperties: false
+    additionalProperties: false
+
+required:
+  - compatible
+  - reg
+  - '#clock-cells'
+  - '#reset-cells'
+  - clocks
+  - clock-names
+  - interrupts
+
+additionalProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/clock/st,stm32mp25-rcc.h>
+
+    rcc: clock-controller@44200000 {
+        compatible = "st,stm32mp25-rcc";
+        reg = <0x44200000 0x10000>;
+        #clock-cells = <1>;
+        #reset-cells = <1>;
+        clock-names = "hse", "hsi", "msi", "lse", "lsi";
+        clocks = <&scmi_clk CK_SCMI_HSE>,
+                 <&scmi_clk CK_SCMI_HSI>,
+                 <&scmi_clk CK_SCMI_MSI>,
+                 <&scmi_clk CK_SCMI_LSE>,
+                 <&scmi_clk CK_SCMI_LSI>;
+
+        interrupts = <GIC_SPI 259 IRQ_TYPE_LEVEL_HIGH>;
+
+        st,c1msrd = <31>;
+        st,flexgen = <
+                FLEXGEN_CFG(0, XBAR_SRC_PLL4, 0, 2)
+                FLEXGEN_CFG((1, XBAR_SRC_PLL4, 0, 5)
+                FLEXGEN_CFG((2, XBAR_SRC_PLL4, 0, 1)
+                FLEXGEN_CFG((3, XBAR_SRC_PLL4, 0, 2)
+                FLEXGEN_CFG((4, XBAR_SRC_PLL4, 0, 3)
+                FLEXGEN_CFG((5, XBAR_SRC_PLL4, 0, 2)
+                FLEXGEN_CFG((6, XBAR_SRC_PLL4, 0, 1)
+                FLEXGEN_CFG((7, XBAR_SRC_PLL4, 0, 11)
+                FLEXGEN_CFG((8, XBAR_SRC_PLL4, 0, 11)
+                FLEXGEN_CFG((9, XBAR_SRC_PLL4, 0, 11)
+                FLEXGEN_CFG((10, XBAR_SRC_PLL4, 0, 11)
+                FLEXGEN_CFG((11, XBAR_SRC_PLL4, 0, 5)
+                FLEXGEN_CFG((12, XBAR_SRC_PLL4, 0, 11)
+                FLEXGEN_CFG((13, XBAR_SRC_PLL4, 0, 11)
+                FLEXGEN_CFG((14, XBAR_SRC_PLL4, 0, 11)
+                FLEXGEN_CFG((15, XBAR_SRC_PLL4, 0, 11)
+                FLEXGEN_CFG((16, XBAR_SRC_PLL4, 0, 5)
+                FLEXGEN_CFG((17, XBAR_SRC_PLL5, 0, 3)
+                FLEXGEN_CFG((18, XBAR_SRC_PLL5, 0, 3)
+                FLEXGEN_CFG((19, XBAR_SRC_PLL5, 0, 3)
+                FLEXGEN_CFG((20, XBAR_SRC_PLL4, 0, 11)
+                FLEXGEN_CFG((21, XBAR_SRC_PLL4, 0, 11)
+                FLEXGEN_CFG((22, XBAR_SRC_PLL4, 0, 11)
+                FLEXGEN_CFG((23, XBAR_SRC_PLL7, 0, 16)
+                FLEXGEN_CFG((24, XBAR_SRC_PLL7, 0, 16)
+                FLEXGEN_CFG((25, XBAR_SRC_PLL7, 0, 16)
+                FLEXGEN_CFG((26, XBAR_SRC_PLL4, 0, 11)
+                FLEXGEN_CFG((27, XBAR_SRC_PLL8, 0, 3)
+                FLEXGEN_CFG((28, XBAR_SRC_PLL8, 0, 21)
+                FLEXGEN_CFG((29, XBAR_SRC_PLL5, 0, 1)
+                FLEXGEN_CFG((30, XBAR_SRC_HSE, 0, 1)
+                FLEXGEN_CFG((31, XBAR_SRC_PLL5, 0, 19)
+                FLEXGEN_CFG((32, XBAR_SRC_PLL5, 0, 19)
+                FLEXGEN_CFG((33, XBAR_SRC_PLL4, 0, 23)
+                FLEXGEN_CFG((34, XBAR_SRC_PLL4, 0, 59)
+                FLEXGEN_CFG((35, XBAR_SRC_HSI, 0, 3)
+                FLEXGEN_CFG((36, XBAR_SRC_PLL5, 0, 3)
+                FLEXGEN_CFG((37, XBAR_SRC_PLL5, 0, 3)
+                FLEXGEN_CFG((38, XBAR_SRC_PLL5, 0, 3)
+                FLEXGEN_CFG((39, XBAR_SRC_PLL5, 0, 3)
+                FLEXGEN_CFG((40, XBAR_SRC_PLL4, 0, 11)
+                FLEXGEN_CFG((41, XBAR_SRC_PLL4, 0, 11)
+                FLEXGEN_CFG((42, XBAR_SRC_PLL7, 0, 6)
+                FLEXGEN_CFG((43, XBAR_SRC_PLL4, 0, 23)
+                FLEXGEN_CFG((44, XBAR_SRC_PLL4, 0, 5)
+                FLEXGEN_CFG((45, XBAR_SRC_PLL4, 0, 2)
+                FLEXGEN_CFG((46, XBAR_SRC_PLL5, 0, 3)
+                FLEXGEN_CFG((47, XBAR_SRC_PLL5, 0, 3)
+                FLEXGEN_CFG((48, XBAR_SRC_PLL5, 0, 3)
+                FLEXGEN_CFG((49, XBAR_SRC_PLL5, 0, 3)
+                FLEXGEN_CFG((50, XBAR_SRC_PLL4, 0, 5)
+                FLEXGEN_CFG((51, XBAR_SRC_PLL4, 0, 5)
+                FLEXGEN_CFG((52, XBAR_SRC_PLL4, 0, 5)
+                FLEXGEN_CFG((53, XBAR_SRC_PLL4, 0, 5)
+                FLEXGEN_CFG((54, XBAR_SRC_PLL6, 0, 3)
+                FLEXGEN_CFG((55, XBAR_SRC_PLL6, 0, 3)
+                FLEXGEN_CFG((56, XBAR_SRC_PLL4, 0, 5)
+                FLEXGEN_CFG((57, XBAR_SRC_HSE, 0, 1)
+                FLEXGEN_CFG((58, XBAR_SRC_HSE, 0, 1)
+                FLEXGEN_CFG((59, XBAR_SRC_PLL4, 0, 1)
+                FLEXGEN_CFG((60, XBAR_SRC_PLL6, 0, 3)
+                FLEXGEN_CFG((61, XBAR_SRC_PLL4, 0, 7)
+                FLEXGEN_CFG((62, XBAR_SRC_PLL4, 0, 7)
+                FLEXGEN_CFG((63, XBAR_SRC_PLL4, 0, 2)
+        >;
+
+        st,busclk = <
+                DIV_CFG(DIV_LSMCU, 1)
+                DIV_CFG(DIV_APB1, 0)
+                DIV_CFG(DIV_APB2, 0)
+                DIV_CFG(DIV_APB3, 0)
+                DIV_CFG(DIV_APB4, 0)
+                DIV_CFG(DIV_APBDBG, 0)
+        >;
+
+        st,kerclk = <
+                DIV_CFG(DIV_RTC, 23)
+                MUX_CFG(MUX_ADC12, MUX_ADC12_FLEX46)
+                MUX_CFG(MUX_ADC3, MUX_ADC3_FLEX47)
+                MUX_CFG(MUX_USB2PHY1, MUX_USB2PHY1_FLEX57)
+                MUX_CFG(MUX_USB2PHY2, MUX_USB2PHY2_FLEX58)
+                MUX_CFG(MUX_USB3PCIEPHY, MUX_USB3PCIEPHY_FLEX34)
+                MUX_CFG(MUX_DSIPHY, MUX_DSIPHY_FLEX28)
+                MUX_CFG(MUX_DSIBLANE, MUX_DSIBLANE_DSIPHY)
+                MUX_CFG(MUX_LVDSPHY, MUX_LVDSPHY_FLEX32)
+                MUX_CFG(MUX_DTS, MUX_DTS_HSE)
+                MUX_CFG(MUX_RTC, MUX_RTC_HSE)
+                MUX_CFG(MUX_D3PER, MUX_D3PER_LSE)
+                MCO_CFG(MCO1, MUX_MCO1_FLEX61, MCO_OFF)
+                MCO_CFG(MCO2, MUX_MCO2_FLEX62, MCO_OFF)
+        >;
+
+        pll1: st,pll-1 {
+                st,pll = <&pll1_cfg_1200Mhz>;
+
+                pll1_cfg_1200Mhz: pll1-cfg-1200Mhz {
+                  cfg = < 30 1 1 1 >;
+                  src = < MUX_CFG(MUX_MUXSEL5, MUXSEL_HSE) >;
+                };
+
+                pll1_cfg_1500Mhz: pll1-cfg-1500Mhz {
+                  cfg = < 75 2 1 1 >;
+                  src = < MUX_CFG(MUX_MUXSEL5, MUXSEL_HSE) >;
+                };
+        };
+
+        pll2: st,pll-2 {
+                st,pll = <&pll2_cfg_600Mhz>;
+
+                pll2_cfg_600Mhz: pll2-cfg-600Mhz {
+                  cfg = < 30 1 1 2 >;
+                  src = < MUX_CFG(MUX_MUXSEL6, MUXSEL_HSE) >;
+                };
+              };
+
+        pll3: st,pll-3 {
+              st,pll = <&pll3_cfg_800Mhz>;
+
+              pll3_cfg_800Mhz: pll3-cfg-800Mhz {
+                cfg = < 20 1 1 1 >;
+                src = < MUX_CFG(MUX_MUXSEL7, MUXSEL_HSE) >;
+              };
+
+              pll3_cfg_900Mhz: pll3-cfg-900Mhz {
+                cfg = < 45 2 1 1 >;
+                src = < MUX_CFG(MUX_MUXSEL7, MUXSEL_HSE) >;
+              };
+        };
+
+        pll4: st,pll-4 {
+              st,pll = <&pll4_cfg_1200Mhz>;
+
+              pll4_cfg_1200Mhz: pll4-cfg-1200Mhz {
+                cfg = < 30 1 1 1 >;
+                src = < MUX_CFG(MUX_MUXSEL0, MUXSEL_HSE) >;
+              };
+        };
+
+        pll5: st,pll-5 {
+              st,pll = <&pll5_cfg_532Mhz>;
+
+              pll5_cfg_532Mhz: pll5-cfg-532Mhz {
+                cfg = < 133 5 1 2 >;
+                src = < MUX_CFG(MUX_MUXSEL1, MUXSEL_HSE) >;
+              };
+        };
+
+        pll6: st,pll-6 {
+              st,pll = <&pll6_cfg_500Mhz>;
+
+              pll6_cfg_500Mhz: pll6-cfg-500Mhz {
+                cfg = < 25 1 1 2 >;
+                src = < MUX_CFG(MUX_MUXSEL2, MUXSEL_HSE) >;
+              };
+        };
+
+        pll7: st,pll-7 {
+              st,pll = <&pll7_cfg_835_51172Mhz>;
+
+              pll7_cfg_835_51172Mhz: pll7-cfg-835-51172Mhz {
+                cfg = < 167 4 1 2 >;
+                src = < MUX_CFG(MUX_MUXSEL3, MUXSEL_HSE) >;
+                frac = < 0x1A3337 >;
+              };
+        };
+
+        pll8: st,pll-8 {
+          st,pll = <&pll8_cfg_594Mhz>;
+
+          pll8_cfg_594Mhz: pll8-cfg-594Mhz {
+                cfg = < 297 5 1 4 >;
+                src = < MUX_CFG(MUX_MUXSEL4, MUXSEL_HSE) >;
+              };
+        };
+
+        st,clk_opp {
+                st,ck_cp1 {
+                    cfg_1 {
+                        hz = < 1500000000 >;
+                        st,pll = <&pll1_cfg_1500Mhz>;
+                    };
+                    cfg_2 {
+                        hz = < 1200000000 >;
+                        st,pll = <&pll1_cfg_1200Mhz>;
+                    };
+                };
+        };
+    };
+...
-- 
2.47.2

